src/database/db_utils.py

```
@@ -0,0 +1,83 @@
+
+    tuples = [_row_tuple_from_dict(r) for r in rows]
+
+    pool = get_global_pool()
+    # Use the pool's connection context to acquire a raw psycopg2 connection
+    with pool.connection() as conn:
+        cur = conn.cursor()
+        try:
+            execute_values(cur, insert_sql, tuples, page_size=page_size)
+            try:
+                rowcount = cur.rowcount
+            except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)